pkgs/applications/video/pyca/default.nix

```
@@ -1,25 +1,70 @@
-{ stdenv, lib, buildPythonApplication, fetchFromGitHub, pycurl, python-dateutil, configobj, sqlalchemy, sdnotify, flask }:
+{ stdenv, lib, buildPythonApplication, python3, fetchFromGitHub, buildNpmPackage }:
 
+let
+  python = python3.override {
+    packageOverrides = self: super: {
+      # pyCA is incompatible with SQLAlchemy 2.0
+      sqlalchemy = super.sqlalchemy.overridePythonAttrs (old: rec {
+        version = "1.4.46";
+        src = self.fetchPypi {
+          pname = "SQLAlchemy";
+          inherit version;
+          hash = "sha256-aRO4JH2KKS74MVFipRkx4rQM6RaB8bbxj2lwRSAMSjA=";
+        };
+      });
+    };
+  };
+
+  frontend = buildNpmPackage rec {
+    pname = "pyca";
+    version = "4.5";
+
+    src = fetchFromGitHub {
+      owner = "opencast";
+      repo = "pyCA";
+      rev = "v${version}";
+      sha256 = "sha256-cTkWkOmgxJZlddqaSYKva2wih4Mvsdrd7LD4NggxKQk=";
+    };
+
+    npmDepsHash = "sha256-0U+semrNWTkNu3uQQkiJKZT1hB0/IfkL84G7/oP8XYY=";
+
+    nativeBuildInputs = [ python ];
+
+    patches = [ ./add-package-json-version.patch ];
+
+    installPhase = ''
+      mkdir -p $out/static
+      cp -R pyca/ui/static/* $out/static/
+    '';
+  };
+
+in
 buildPythonApplication rec {
   pname = "pyca";
-  version = "2.1";
+  version = "4.5";
 
   src = fetchFromGitHub {
     owner = "opencast";
     repo = "pyCA";
     rev = "v${version}";
-    sha256 = "0cvkmdlcax9da9iw4ls73vw0pxvm8wvchab5gwdy9w9ibqdpcmwh";
+    sha256 = "sha256-cTkWkOmgxJZlddqaSYKva2wih4Mvsdrd7LD4NggxKQk=";
   };
 
-  propagatedBuildInputs = [
+  propagatedBuildInputs = with python.pkgs; [
     pycurl
     python-dateutil
     configobj
     sqlalchemy
     sdnotify
+    psutil
     flask
+    prometheus-client
   ];
 
+  postPatch = ''
+    sed -i -e 's#static_folder=.*#static_folder="${frontend}/static")#' pyca/ui/__init__.py
+  '';
+
```

```suggestion

  passthru = {
    inherit frontend;
  };

```

Allows building `pyCA.frontend` to iterate on just the frontend.

---

Didn't even know about that. Fixed!